### PR TITLE
Added coverage reporting

### PR DESCRIPTION
**Pull request by jjbailey-io**
- Added code coverage reporting
- Updated tests to use faster runners
- Updated workflows to use a consistent version of foundry

### CODE REVIEW

**Review by VoR0220 — APPROVED** [review]

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+          forge install dapphub/ds-test 
+          forge install OpenZeppelin/openzeppelin-contracts
+      - name: Build the repository
+        shell: bash
+        run: |
+          forge clean && forge build --optimize
+      - name: Run tests with coverage option
+        env:
+          ETHEREUM_RPC_KEY: ${{ secrets.ALCHEMY_KEY }}
+          POLYGON_RPC_KEY: ${{ secrets.POLYGON_ALCHEMY_KEY }}
```
Comment (TJ-Everett) [inline .github/workflows/coverage.yml]: Are these test coverage reports utilizing the alchemy key at all? Seems like this wouldn't be ideal if they are as that would be increasing usage on the alchemy account?

**Review by TJ-Everett — APPROVED** [review]
Approved. Looks great, just a question about the Alchemy keys referenced

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (VoR0220) [reply .github/workflows/coverage.yml]: Unless we were running tests off of a forked alchemy environment.

**Review by Palmerg4 — APPROVED** [review]

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (jjbailey-io) [reply .github/workflows/coverage.yml]: Oops... yeah, that's what you get for copy/pasting from other repos. v2 will have this issue too.

**Review by ShaneDuncan602 — APPROVED** [review]
Looks good to me.

**Comment by VoR0220** [comment]
tests failing. Uh oh. @jjbailey-io did you cancel these?

**Pull request merged**